cfn-resources/project/cmd/resource/resource.go

```diff
@@ -86,12 +83,20 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return progressevent.GetFailedEventByResponse(fmt.Sprintf("Failed to Create Project : %s", err.Error()),
 			res), nil
 	}
+	projectID := project.GetId()
 
 	// Add ApiKeys
 	if len(currentModel.ProjectApiKeys) > 0 {
 		for _, key := range currentModel.ProjectApiKeys {
-			_, res, err := atlasV2.ProgrammaticAPIKeysApi.UpdateApiKeyRoles(context.Background(), *project.Id, *key.Key, &admin.UpdateAtlasProjectApiKey{
-				Roles: key.RoleNames,
+			if key.Key == nil {
+				errorMessage := fmt.Sprintf("ApiKey is missing the configuration for projectID=%s", projectID)
+				_, _ = logger.Warn(errorMessage)
```

Comment:
i would remove all logs that don't add anything, like this one. if we're already returning the error then no need to also log it.  i would search for "logger." and see if any one if needed at all

Reply:
Removed all log statements